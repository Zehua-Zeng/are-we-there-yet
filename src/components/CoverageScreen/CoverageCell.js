import React, { useEffect } from "react";
// styled components
import styled from 'styled-components';

const StyledCellContainer = styled.div`
    .grid-values {
        border: solid 0.5px white;
        height: 2.4rem;
    }
    
    .active-value {
        cursor: pointer;
        color: black;
        padding: 7px;
    }

    .coverage-experiment-active {
        background-color: rgb(77, 171, 214);
    }

    .coverage-experiment-deactive {
        cursor: pointer;
        color: lightblue;
        background-color: lightblue;
    }

    .coverage-experiment-deactive:hover {
        color: #70C2E8;
        background-color: #70C2E8;
    }

    .coverage-hybrid-active {
        background-color: rgb(141, 204, 108);
    }

    .coverage-hybrid-deactive {
        color: rgb(181, 237, 152);
        background-color: rgb(181, 237, 152);
    }

    .coverage-hybrid-deactive:hover {
        color: rgb(162, 222, 131);
        background-color: rgb(162, 222, 131);
    }

    .coverage-theory-active {
        background-color: rgb(235, 196, 96);
    }

    .coverage-theory-deactive {
        cursor: pointer;
        color: rgb(250, 224, 152);
        background-color: rgb(250, 224, 152);
    }

    .coverage-theory-deactive:hover {
        color: rgb(245, 210, 118);
        background-color: rgb(245, 210, 118);
    }

    .coverage-none {
        color: lightgrey;
        background-color: lightgrey;
    }

    .coverage-none:hover {
        color: #70C2E8;
        background-color: #70C2E8;
    }

    .coverage-none-active {
        background-color: rgb(77, 171, 214);
    }

    .coverage-none-deactive {
        cursor: pointer;
        color: lightgrey;
        background-color: lightgrey;
    }

    .coverage-none-deactive:hover {
        color: #70C2E8;
        background-color: #70C2E8;
    }

`;

var CoverageCell = ({category, name, handleClick, status, isActive, type}) => {
    const [active, setActive] = React.useState(status);

    let clickEvent = () => {
        setActive(!active);
        handleClick(!active, name);
    }

    useEffect(() => {
        setActive(status);
    }, [status]);

    if (category.localeCompare("coverage-none") === 0) {
        // filter matrix cell have states depends on parent component
        if (type === 'filter') {
            return (
                <StyledCellContainer>
                    <div className={`grid-values ${isActive ? "coverage-none-active active-value" : "coverage-none-deactive"}`} onClick={() => handleClick()}>
                        {isActive ? <i className="fas fa-check"></i> : <p>X</p>}
                    </div>
                </StyledCellContainer>
                
            );
        }

        return (<StyledCellContainer>
        {/* <div className={`grid-values ${category} active-value`} onClick={() => clickEvent()}>
            X
        </div> */}
        <div className={`grid-values ${active ? "coverage-none-active active-value" : "coverage-none-deactive"}`} onClick={() => clickEvent()}>
            {active ? <i className="fas fa-check"></i> : <p>X</p>}
        </div>
    </StyledCellContainer>);
    }
    
    return (
        <StyledCellContainer key={name}>
            <div className={`grid-values ${active ? category+"-active active-value" : category+"-deactive"}`} onClick={() => clickEvent()}>
               {active ?  <i className="fas fa-check"></i> : <p>X</p> }
            </div>

        </StyledCellContainer>
    );
   
}

export default CoverageCell;