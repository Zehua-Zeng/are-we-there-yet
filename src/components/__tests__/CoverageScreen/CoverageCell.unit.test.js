import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import CoverageCell from '../../CoverageScreen/CoverageCell';
import renderer from 'react-test-renderer';
// Do not remove, used in the file
import {toBeInTheDocument} from '@testing-library/jest-dom';

afterEach(() => {
    cleanup();
});

test('should render unselected cell matches snapshot', () => {
    // key={`${t}-2`} category='coverage-none' name={t} 
    // handleClick={() => handleClick(1,t)} 
    // isActive={selectedFirstEncoding === t} type="filter"
    let key = 'somekey';
    let category = 'coverage-none';
    let name = "PX";
    let handleClick= () => {};
    let isActive = false;
    let type = "filter";

    let cell = <CoverageCell key={key} category={category} name={name} handleClick={handleClick} isActive={isActive} type={type}/>;
    render(cell);

    const cellElement = screen.getByTestId(name);
    // expect it to be rendered in DOM
    expect(cellElement).toBeInTheDocument();

    const tree = renderer.create(cell).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should render selected cell matches snapshot', () => {
    let key = 'somekey';
    let category = 'coverage-none';
    let name = "PY";
    let handleClick= () => {};
    let isActive = true;
    let type = "filter";

    let cell = <CoverageCell key={key} category={category} name={name} handleClick={handleClick} isActive={isActive} type={type}/>;
    render(cell);

    const cellElement = screen.getByTestId(name);
    // expect it to be rendered in DOM
    expect(cellElement).toBeInTheDocument();

    const tree = renderer.create(cell).toJSON();
    expect(tree).toMatchSnapshot();
});