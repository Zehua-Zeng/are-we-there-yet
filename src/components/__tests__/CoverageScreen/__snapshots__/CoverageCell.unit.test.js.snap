// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render selected cell matches snapshot 1`] = `
<div
  className="sc-bdnxRM lmmjcq"
  data-testid="PY"
  id="PY-undefined"
>
  <div
    className="grid-values coverage-none-active active-value"
    onClick={[Function]}
  >
    <i
      className="fas fa-check"
    />
  </div>
</div>
`;

exports[`should render unselected cell matches snapshot 1`] = `
<div
  className="sc-bdnxRM lmmjcq"
  data-testid="PX"
  id="PX-undefined"
>
  <div
    className="grid-values coverage-none-deactive"
    onClick={[Function]}
  >
    <p>
      X
    </p>
  </div>
</div>
`;
